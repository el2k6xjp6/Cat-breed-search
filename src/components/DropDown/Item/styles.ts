import styled from "styled-components";

export const ListItem = styled.li`
  display: flex;
  align-items: center;
  border-radius: 4px;
  font-size: 16px;
  height: 36px;
  line-height: 36px;
  list-style-type: none;
  padding: 0 12px;
  cursor: pointer;
  :hover {
    background: #dddddd;
  }
`;

export const ListItemText = styled.div`
  @media (hover: hover) and (pointer: fine) {
    ${ListItem}: hover & {
      color: black;
    }
  }
`;
