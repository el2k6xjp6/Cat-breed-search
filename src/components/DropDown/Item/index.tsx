import React from "react";
import { ListItem, ListItemText } from "../styles";

export type optionType = {
  text: string;
  value: number;
};

type DropDownListProps = {
  option: optionType;
  index: number;
  onOptionClicked: (option: optionType) => void;
};

const Item = ({ option, index, onOptionClicked }: DropDownListProps) => {
  const { text, value } = option;
  return (
    <ListItem onClick={() => onOptionClicked(option)} key={index} value={value}>
      <ListItemText>{text}</ListItemText>
    </ListItem>
  );
};

export default Item;
