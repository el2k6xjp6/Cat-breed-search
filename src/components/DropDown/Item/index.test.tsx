import { shallow } from "enzyme";
import Item from ".";
import "jest-styled-components";
import renderer from "react-test-renderer";

describe("test DropDown", () => {
  const mockProps = {
    option: { text: "test", value: 0 },
    index: 0,
    onOptionClicked: jest.fn(),
  };

  test("Snapshot for DropDown", () => {
    const tree = renderer.create(<Item {...mockProps} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  test("toggle", () => {
    const wrapper = shallow(<Item {...mockProps} />);
    wrapper.find("[id='option 0']").simulate("click");
    expect(mockProps.onOptionClicked).toBeCalled();
  });
});

export {};
