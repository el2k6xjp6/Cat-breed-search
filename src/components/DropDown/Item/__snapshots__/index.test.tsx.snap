// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test DropDown Snapshot for DropDown 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border-radius: 4px;
  font-size: 16px;
  height: 36px;
  line-height: 36px;
  list-style-type: none;
  padding: 0 12px;
  cursor: pointer;
}

.c1:hover {
  background: #dddddd;
}

@media (hover:hover) and (pointer:fine) {
  .c0:hover .c2 {
    color: black;
  }
}

<li
  className="c0 c1"
  id="option 0"
  onClick={[Function]}
  value={0}
>
  <div
    className="c2 "
  >
    test
  </div>
</li>
`;
